### PR TITLE
More Documentation

### PR DESCRIPTION
**Pull request by HilbertCurve**
I also fixed that inconsistency in the class documentation of Scene.java.

### CODE REVIEW

File: src/main/java/event/EventListener.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
 package event;
 
 /**
- * A Simple listener interface
+ * <h1>Azurite</h1>
+ * A Simple listener interface.
+ * Any class that implements this "subscribes" to the {@code EventData}
+ * class put in the type parameter. Whenever such an event is triggered,
+ * {@code this.onEvent(triggeredEvent)} is ran.
```
Comment (athaun) [inline src/main/java/event/EventListener.java:8]: Slight grammar nitpick, it should be "run" not ran.

File: src/main/java/graphics/Shader.java [hunk 2]
```diff
@@ -12,6 +12,24 @@
 import static org.lwjgl.opengl.GL11.GL_FALSE;
 import static org.lwjgl.opengl.GL20.*;
 
+/**
+ * <h1>Azurite</h1>
+ * A shader is a critical component in the OpenGL rendering process.
+ * The GPU uses a shader as a "program" which tells the GPU how to
+ * shade pixels, fragments, and vertices. A shader comes in two parts:
```
Comment (athaun) [inline src/main/java/graphics/Shader.java:19]: Suggestion to change "how to shade..." to "how to color..."

File: src/main/java/graphics/Texture.java [hunk 3]
```diff
@@ -17,6 +17,7 @@
 import static org.lwjgl.stb.STBImage.*;
 
 /**
+ * <h1>Azurite</h1>
```
Comment (athaun) [inline src/main/java/graphics/Texture.java:20]: These Azurite headers seem unnecessary. If people are reading the documentation, they will already be aware that it is documentation for Azurite.

File: src/main/java/graphics/renderer/QuadRenderBatch.java [hunk 4]
```diff
@@ -5,6 +5,16 @@
 import graphics.ShaderDatatype;
 import graphics.Window;
 
+/**
+ * <h1>Azurite</h1>
+ * Used to render one large rectangle that stretches to all corners
+ * of the window. This can be used in shaders to apply effects based
+ * on the position inside this rectangle, such as point-light effects.
+ *
```
Comment (athaun) [inline src/main/java/graphics/renderer/QuadRenderBatch.java:13]: The quad renderer is used for more than just a single quad across the window. As the name of this class implies, it is able to handle batch rendering for many quads. Every texture that is rendered is rendered on a quad, as well as any rectangle primitives.

File: src/main/java/input/Gamepad.java [hunk 5]
```diff
@@ -4,6 +4,10 @@
 
 // https://github.com/LWJGL/lwjgl3-wiki/wiki/2.6.3-Input-handling-with-GLFW#joystick-input
 
+/**
+ * <h1>Azurite</h1>
+ * Abstraction over LWJGL specification of gamepad buttons.
```
Comment (athaun) [inline src/main/java/input/Gamepad.java:9]: More specifically, the API we are making use of is provided by GLFW, not just LWGJL.

File: src/main/java/input/Mouse.java [hunk 6]
```diff
@@ -10,6 +10,11 @@
 
 import static org.lwjgl.glfw.GLFW.*;
 
+/**
+ * <h1>Azurite</h1>
+ * Abstraction over LWJGL specification of mouse scroll,
```
Comment (athaun) [inline src/main/java/input/Mouse.java:15]: Same as my previous comment, these methods are provided by GLFW.

File: src/main/java/scene/Scene.java [hunk 7]
```diff
@@ -59,9 +59,22 @@
  *         public void awake() {
  *             camera = new Camera();
  *
+ *             // this is the player //
```
Comment (athaun) [inline src/main/java/scene/Scene.java:62]: In this case, the comment is not necessary... the name of the variable explains the purpose of the object.
image

**Review by athaun — CHANGES_REQUESTED** [review]

File: src/main/java/graphics/renderer/QuadRenderBatch.java (hunk 4 above)
Comment (HilbertCurve) [reply src/main/java/graphics/renderer/QuadRenderBatch.java:13]: I thought the render batch used for multiple quads is DefaultRenderBatch; when looking in the constructor of QuadRenderBatch it called the super constructor with `maxBatchSize = 1`. Maybe I'm missing something?

**Review by athaun — APPROVED** [review]

**Comment by athaun** [comment]
The changes look good, as soon as I can I will resolve the conflicts and merge it.

**Comment by PixelRifts** [comment]
This one's old, I resolved conflicts. 2 lines will still have to be changed after merging renderer refactor

**Comment by athaun** [comment]
> This one's old, I resolved conflicts. 2 lines will still have to be changed after merging renderer refactor

Awesome, I already approved it for merge a while back so u can whenever or I will tmrw if you are good with that.

**Pull request closed without merging**